samples/subsys/video/capture_to_lcd/src/main.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2024 Charles Dias <charlesdias.cd@outlook.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/video.h>
+#include <lvgl.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+#define VIDEO_DEV_SW "VIDEO_SW_GENERATOR"
+
+void draw_border_and_label(void)
+{
+	/*Create an array for the border points */
+	static lv_point_t line_points[] = { {4, 0}, {156, 0}, {156, 72}, {4, 72}, {4, 0} };
+
+	/*Create style*/
+	static lv_style_t style_line;
+
+	lv_style_init(&style_line);
+	lv_style_set_line_width(&style_line, 4);
+	lv_style_set_line_color(&style_line, lv_palette_main(LV_PALETTE_RED));
+	lv_style_set_line_rounded(&style_line, true);
+
+	/*Create a line and apply the new style */
+	lv_obj_t *border = lv_line_create(lv_scr_act());
+	int quant_points = ARRAY_SIZE(line_points);
+
+	lv_line_set_points(border, line_points, quant_points);
+	lv_obj_add_style(border, &style_line, 0);
+	lv_obj_align(border, LV_ALIGN_BOTTOM_LEFT, 0, 0);
+
+	lv_obj_t *label = lv_label_create(border);
+
+	lv_label_set_text(label, "LVGL on Zephyr!");
+	lv_obj_align(label, LV_ALIGN_BOTTOM_MID, 0, -10);
+
+	lv_task_handler();
+}
+
+void draw_lvgl_logo(void)
+{
+	LV_IMG_DECLARE(img_lvgl_logo);
+	lv_obj_t *img1 = lv_img_create(lv_scr_act());
+
+	lv_img_set_src(img1, &img_lvgl_logo);
+	lv_obj_align(img1, LV_ALIGN_TOP_MID, 0, 10);
+
+	lv_task_handler();
+}
+
+void display_splash_screen(void)
+{
+	draw_border_and_label();
+	draw_lvgl_logo();
+
+	k_sleep(K_MSEC(5000));
+	lv_obj_clean(lv_scr_act());
+}
+
+int main(void)
+{
+	const uint16_t WIDTH_VIDEO = CONFIG_VIDEO_WIDTH;
+	const uint16_t HEIGHT_VIDEO = CONFIG_VIDEO_HEIGHT;
+	const struct device *display_dev;
+	struct video_format fmt;
+	struct video_caps caps;
+	const struct device *video;
+	unsigned int frame = 0;
+	size_t bsize;
+	int i = 0;
+
+	display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device not ready, aborting test");
+		return 0;
+	}
+
+	/* Default to software video pattern generator */
+	video = device_get_binding(VIDEO_DEV_SW);
```

Comment:
This is present in the original [video sample]() but the preferred API changed since then it seems?
https://github.com/zephyrproject-rtos/zephyr/blob/55c8a1650235cbf26145160b149a7f6bce2954eb/doc/releases/migration-guide-3.6.rst#devicetree-labels
https://github.com/zephyrproject-rtos/zephyr/blob/55c8a1650235cbf26145160b149a7f6bce2954eb/doc/releases/release-notes-3.2.rst#devicetree

Reply:
Considering that the goal here is to display the frames capture by a real camera, I can remove the code below:

```
	video = device_get_binding(VIDEO_DEV_SW);
	if (video == NULL) {
		LOG_ERR("Video device %s not found", VIDEO_DEV_SW);
		return 0;
	}
```

Reply:
As see [here](https://github.com/zephyrproject-rtos/zephyr/blob/8ae0934470c9cd2bc56f4f5c4308c8b3f59485e9/drivers/video/Kconfig.sw_generator#L7-L9), there is no [safeguard](https://github.com/zephyrproject-rtos/zephyr/blob/8ae0934470c9cd2bc56f4f5c4308c8b3f59485e9/drivers/video/Kconfig.ov2640#L9) to check that the SW generator is on the device tree.  This is understandable as it was not updated since 5 years ago, where API might have been different.

This looks like lacking something like a device-tree `chosen` block to make the example generic, and then using [`DT_CHOSEN()`](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_CHOSEN) here?

```c
video = DT_CHOSEN(camera);
```

These are all just suggestions: I am by no way endorsed by Zephyr in any way. Just offering feedback. :)

Reply:
I appreciate your feedback! I'll test this suggestion once the required PRs are merged. Thanks!

Reply:
This also needs another pull request (I will try to make one) to add a `#define DT_DRV_COMPAT` and YAML binding so that the video pattern generator can be used on a devicetree, so best to keep it as you did until then I think.